tests/test_project.py

```diff
@@ -442,6 +447,39 @@ def op2(job):
         assert len(B._collect_postconditions()[op2]) == 2
         assert len(C._collect_postconditions()[op2]) == 3
 
+    def test_operations_user_error_handling(self):
+        class A(FlowProject):
+            pass
+
+        @A.operation
+        def test_basic_op_error(job):
+            raise Exception
+
+        @A.operation
+        @cmd
+        def test_cmd_op_error(job):
+            raise Exception
+
+        @A.operation
+        @with_job
+        def test_with_job_op_error(job):
+            raise Exception
+
+        @A.operation
+        @with_job
+        @cmd
+        def test_with_job_cmd_error(job):
+            raise Exception
+
+        project = self.mock_project(A)
+        with add_cwd_to_environment_pythonpath():
+            with switch_to_directory(project.root_directory()):
+                with pytest.raises(UserOperationError):
+                    starting_dir = os.getcwd()
+                    with redirect_stderr(StringIO()):
+                        project.run()
```

Comment:
This should stop after the first error. In other words, the other operations would not be tested. You can either use the `operation` argument in `run` or create multiple tests.

In other words, leave the two tests below and just change the error type for the `cmd` one.